### PR TITLE
Make plugin CI ready

### PR DESCRIPTION
**Pull request by navidshaikh**
- Update go module path to `knative.dev/kn-plugin-source-kafkfa` (earlier `client-contrib`..)
- Update and refactor CI scripts
- Add hack scripts
- Remove test-infra 
- Update hack/build.sh script 
- Update build to use `-mod=vendor`
- Add codecov (.codecov.yaml)

### CODE REVIEW

File: docs/cmd/kn.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+## kn
```
Comment (rhuss) [inline docs/cmd/kn.md:1]: How could this sneak in ? We need to ensure that the documentation generated is only for the plugin itself

**Review by rhuss — APPROVED** [review]
Looks good in general but I wonder how the full documentation of kn could get in ?

I suggest to merge this PR and then fine tune it in future steps
/lgtm

File: docs/cmd/kn.md (hunk 1 above)
Comment (navidshaikh) [reply docs/cmd/kn.md:1]: /hold

looks like `hack/build.sh` needs some tweaks

**Review by navidshaikh — COMMENTED** [review]
/hold

File: test/common.sh [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+source $(dirname $0)/../vendor/knative.dev/hack/e2e-tests.sh
+
+function cluster_setup() {
+  header "Building plugin"
+  ${REPO_ROOT_DIR}/hack/build.sh -f || return 1
```
Comment (daisy-ycguo) [inline test/common.sh:21]: Maybe we need to install kn client while cluster is setting up.
Refer to https://github.com/knative/client-contrib/blob/master/test/common.sh#L67

**Comment by navidshaikh** [comment]
```
2020/11/26 07:52:50 No build found for bucket 'knative-prow' and object 'logs/post-knative-sandbox-kn-plugin-source-kafka-go-coverage'
```
seeing go-coverage job complaining ^ above (ref [logs](https://prow.knative.dev/view/gs/knative-prow/pr-logs/pull/knative-sandbox_kn-plugin-source-kafka/4/pull-knative-sandbox-kn-plugin-source-kafka-go-coverage/1331868145438167040#1:build-log.txt%3A754)) does this need any update in test-infra configs etc? cc: @chizhg @coryrc

**Comment by navidshaikh** [comment]
I've added https://github.com/knative/test-infra/pull/2592 for the failing go-coverage job
cc: @rhuss @chizhg

**Review by daisy-ycguo — APPROVED** [review]
LGTM. 
I agree we disable go-coverage test for now and work on it in the future.

**Comment by navidshaikh** [comment]
/close

The go-coverage prow job still appears in the checks list above though, will re-open again to refresh.

**Pull request closed without merging**